src/v/container/include/container/fragmented_vector.h

```diff
@@ -190,9 +208,52 @@ class fragmented_vector {
     size_t size() const noexcept { return _size; }
 
     void shrink_to_fit() {
-        if (!_frags.empty()) {
-            _frags.back().shrink_to_fit();
+        // Calling shrink to fix then modifying the container could result in
+        // allocations that overshoot the template parameter.
+        //
+        // if (!_frags.empty()) {
+        //    _frags.back().shrink_to_fit();
+        // }
+    }
+
+    void reserve(size_t v) {
```

Comment:
Can you document this public method?

On vector reserve() has at least two purposes; (a) performance and (b) iterator and pointer stability: if you reserve space for N elements it is guaranteed that you can push back N elements without breaking pointer stability. 

If have the same or different or no guarantee around that we should document it.

We should document what it does on the performance size too.

Reply:
I think to make things easier conceptually (especially for new folks) we should make the exact same guarantee as `std::vector`. To me this class should be "std::vector" but under the hood doesn't have oversized allocs.